src/webSockets.re

```diff
@@ -1,114 +1,112 @@
+  [@bs.get] external code : t => int = "";
+  [@bs.get] external reason : t => string = "";
+  [@bs.get] external wasClean : t => bool = "";
 };
 
 module type WebSocketMaker = {
   type t;
   let make: string => t;
-  let makeWithProtocols: string => protocols::'a => t;
+  let makeWithProtocols: (string, ~protocols :'a) => t;
 };
 
-module MakeWebSocket (Maker: WebSocketMaker) => {
+module MakeWebSocket (Maker: WebSocketMaker) {
   type t = Maker.t;
   let make = Maker.make;
-  let makeWithProtocol url protocol::(p: string) => Maker.makeWithProtocols url protocols::p;
-  let makeWithProtocols url protocols::(ps: list string) =>
-    Maker.makeWithProtocols url protocols::(Array.of_list ps);
+  let makeWithProtocol = (url, ~protocol: string) => Maker.makeWithProtocols(url, ~protocols=[protocol]);
+  let makeWithProtocols = (url, ~protocols: list(string)) => Maker.makeWithProtocols(url, ~protocols=(Array.of_list(protocols)));
   type binaryType =
     | Blob
     | ArrayBuffer;
-  external _binaryType : t => string = "binaryType" [@@bs.get];
-  external _setBinaryType : t => string => unit = "binaryType" [@@bs.set];
-  let binaryType t => {
-    let str = _binaryType t;
+  [@bs.get] external _binaryType : t => string = "binaryType";
+  [@bs.set] external _setBinaryType : t => string => unit = "binaryType";
+  let binaryType = t => {
+    let str = _binaryType(t);
     str == "blob" ? Blob : ArrayBuffer
   };
-  let setBinaryType binaryType t => {
+  let setBinaryType = (binaryType, t) => {
     let typestr =
       binaryType |> (
         fun
         | Blob => "blob"
         | ArrayBuffer => "arraybuffer"
       );
-    _setBinaryType t typestr;
+    _setBinaryType(t, typestr);
     t
   };
-  external bufferedAmount : t => int64 = "" [@@bs.get];
-  external extensions : t => 'a = "" [@@bs.get];
+  [@bs.get] external bufferedAmount : t => int64 = "";
+  [@bs.get] external extensions : t => 'a = "";
   type event =
     | Close (CloseEvent.t => unit)
     | Error (string => unit)
     | Message (MessageEvent.t => unit)
     | Open (unit => unit);
-  external _on : string => (Js.t {..} => unit) => unit = "addEventListener" [@@bs.send.pipe : t];
-  let on e t => {
+  [@bs.send.pipe : t] external _on : string => (Js.t({.}) => unit) => unit = "addEventListener";
+  let on = (e, t) => {
     let evtname =
       switch e {
-      | Close _ => "close"
-      | Error _ => "error"
-      | Message _ => "message"
-      | Open _ => "open"
+      | Close(_) => "close"
+      | Error(_) => "error"
+      | Message(_) => "message"
+      | Open(_) => "open"
       };
-    _on
-      evtname
-      (
-        fun jsobj =>
+    _on(
+      evtname,
+      ((jsobj: Js.t({.})) =>
           switch e {
-          | Close fn => fn @@ Obj.magic jsobj
-          | Error fn => fn jsobj##message
-          | Message fn => fn @@ Obj.magic jsobj
-          | Open fn => fn ()
+          | Close(fn) => fn(Obj.magic(jsobj))
+          | Error(fn) => fn(Obj.magic(jsobj))
```

Comment:
`Obj.magic` breaks the type system, and the only reason this compiles is because of that. It will fail at runtime. Unless you know exactly what you're doing, you should avoid `Obj.magic` like the plague. This should be `fn(jsobj##message)` (the only thing that's changed is the function application syntax)